Added test that no files are executable

ramvenkat98 opened the pull request:
<!-- Your title above should be a short description of what
was changed. Do not include the issue number in the title. -->

#### References to other Issues or PRs
<!-- If this pull request fixes an issue, write "Fixes #NNNN" in that exact
format, e.g. "Fixes #1234". See
https://github.com/blog/1506-closing-issues-via-pull-requests .-->
Fixes #14845 

#### Brief description of what is fixed or changed
Created test_executable.py which tests that only files with shebangs have
an executable bit set. These two files violated this and had permissions
changed (to code number 644):
    sympy/utilities/tests/test_module_imports.py
    data/TeXmacs/progs/init-sympy.scm

#### Other comments

#### Release Notes

<!-- Write the release notes for this release below. See
https://github.com/sympy/sympy/wiki/Writing-Release-Notes for more information
on how to write release notes. If there is no release notes entry for this PR,
write "NO ENTRY". The bot will check your release notes automatically to see
if they are formatted correctly. -->

<!-- BEGIN RELEASE NOTES -->
NO ENTRY
<!-- END RELEASE NOTES -->

ramvenkat98 committed:
Added test that no files are executable

Created test_executable.py which tests that only files with shebangs have
an executable bit set. These two files violated this and had permissions
changed (to code number 644):
    sympy/utilities/tests/test_module_imports.py
    data/TeXmacs/progs/init-sympy.scm

ramvenkat98 committed:
Changed permissions of executable files correctly

Changed permissions of these two files to 644:
    sympy/utilities/tests/test_module_imports.py
    data/TeXmacs/progs/init-sympy.scm

asmeurer commented on bin/test_executable.py:
```diff
@@ -0,0 +1,23 @@
+"""
```
Make this file itself executable.

asmeurer commented:
You should add this to the Travis tests. The best place would be [here](https://github.com/sympy/sympy/blob/b70b86b2b948f500782b38f2a20524aa2205a815/bin/test_travis.sh#L36).

asmeurer commented on bin/test_executable.py:
```diff
@@ -0,0 +1,23 @@
+        if os.access(path, os.X_OK):
+            with open(path, 'r') as f:
+                if f.readline()[:2] != "#!":
+                    exn_msg = "Executable bit set for file path" + path
```
Note in the error that the file either should not be executable, or it should have a shebang line.

ramvenkat98 committed:
Corrected test script and added to travis

Made test script executable and added shebang line
Made the error message for failing the test clearer
Added the file to the tests in travis

ramvenkat98 commented:
I've made the modifications suggested by the review and updated the pull request, namely:

1. Making the test file executable by adding a shebang line and setting the permission to 755
2. Added the test file to Travis builds
3. Made the exception message clearer
